azureLogIngestion/index.ts

```diff
@@ -1,21 +1,25 @@
 import { AzureFunction, Context } from "@azure/functions"
-
 import Adapter from "./adapter"
 import OpenTelemetryAdapter from "./opentelemetry"
+import { parse } from "./utils/resource"
 
 const apiKey = process.env["NEW_RELIC_INSERT_KEY"]
 const debug = process.env["DEBUG"] || false
 const otel = process.env["OTEL"] || false
 
 const adapter = new Adapter(apiKey)
-const otelAdapter = new OpenTelemetryAdapter(apiKey)
+let otelAdapter: OpenTelemetryAdapter
 
 const eventHubTrigger: AzureFunction = async function (context: Context, eventHubMessages: any): Promise<void> {
     if (debug) {
         context.log(`Eventhub trigger function called for message eventHubMessages ${eventHubMessages}`)
         context.log(`eventHubMessages type: ${typeof eventHubMessages}`)
     }
     if (otel) {
+        const { resourceId } = eventHubMessages.records[0]
```

Comment:
First, I wouldn't do it this way. This function is going to get records for all sorts of resources. So this won't be reliable.

Second, why are we parsing this value and not `resourceId`? That's what `parse()` is expecting...

Reply:
resourceId isn't available to us yet--if we're going to get it here at all, it's going to be buried in eventHubMessages. So yeah, this isn't optimal. I do agree that we may have to extend traceProvider to let us set this later.